### PR TITLE
fix(blog): apply trailing slash to blog feed

### LINKED ISSUE #9829: Add a trailing slash to feed URLs if `trailingSlash` is enabled
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

Add the trailing slash to URLs generated in the feed files. 

### Has this been requested on Canny?

_No response_

### Motivation

The `trailingSlash` configuration is available to customise the presence/absence of a trailing slash at the end of URLs; unfortunately this setting does not apply to the links generated in the feed files, and tools like Algolia that use the feed files may behave unusual.

### API design

Call the `applyTrailingSlash` function conditionally to adjust the URLs generated in the feed files.

### Have you tried building it?

The functionality is implemented in  https://github.com/cronica-it/docusaurus-chronology/commit/35fef9608ae3014c3487918ad2ef62c5cf273d40.

### Self-service

- [X] I'd be willing to contribute this feature to Docusaurus myself.

### PR DESCRIPTION
**Pull request by OzakIOne**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

tools like Algolia that use the feed files may behave unusual if trailing slash is enabled and not applied to links

## Test Plan

`feed.test.ts` => `has feed item for each post trailing slash`

Deploy preview ??

### Test links

[rss.xml](https://deploy-preview-9920--docusaurus-2.netlify.app/blog/rss.xml)

## Related issues/PRs

Fix #9829

### CODE REVIEW

**Comment by OzakIOne** [comment]
Found a todo
https://github.com/facebook/docusaurus/blob/d7a7caec6d758eabc332e7e697a63e7c85279076/packages/docusaurus-utils-common/src/applyTrailingSlash.ts#L27-L30
I think we can do it in this PR however should we keep the one in `utils-common` or `utils` ?

**Comment by slorber** [comment]
- `utils` is Node.js only utils
- `utils-common` is Browser + Node.js

This code is env agnostic so it should rather be in utils-common, but is not because this shared package was introduced later. I'd prefer if you did this refactor in another PR because this will impact various files unrelated to this feature.

File: packages/docusaurus-plugin-content-blog/src/feed.ts [hunk 1]
```diff
@@ -105,13 +114,19 @@ async function defaultCreateFeedItems({
...
-        permalink.replace(siteConfig.baseUrl, ''),
+        permalink.replace(baseUrl, ''),
         outDir,
-        siteConfig.trailingSlash,
+        trailingSlash,
       );
       const $ = cheerioLoad(content);
 
-      const blogPostAbsoluteUrl = normalizeUrl([siteUrl, permalink]);
+      const blogPostAbsoluteUrl = applyTrailingSlash(
```
Comment (slorber) [inline packages/docusaurus-plugin-content-blog/src/feed.ts:123]: LGTM 👍 

We need a test in `feed.test.ts` covering this, maybe copy/paste `it('has feed item for each post'` and add the site trailing slash config?

**Review by slorber — CHANGES_REQUESTED** [review]

**Review by slorber — APPROVED** [review]

**Comment by ilg-ul** [comment]
Thank you!

**Pull request merged**